armory/data/datasets.py

```diff
@@ -269,6 +269,72 @@ def parse_split_index(split: str):
     return "+".join(output_tokens)
 
 
+def filter_by_index(dataset: "tf.data.Dataset", index: list, dataset_size: int):
+    """
+    index must be a list or iterable of integer values
+
+    returns the dataset and the indexed size
+    """
+    dataset_size = int(dataset_size)
+    index = sorted([int(x) for x in set(index) if int(x) < dataset_size])
+    indexed_size = len(index)
+    if indexed_size == 0:
+        raise ValueError("index must have at least one value")
+    elif index[0] < 0:
+        raise ValueError("index values must be nonnegative")
+
+    index_tensor = tf.constant(index, dtype=tf.int64)
+
+    def enum_index(i, x):
+        i = tf.expand_dims(i, 0)
+        out, _ = tf.raw_ops.ListDiff(x=i, y=index_tensor, out_idx=tf.int64)
+        return tf.size(out) == 0
```

Comment:
```suggestion
        return tf.equal(tf.size(out), 0)
```

Reply:
I think this is the compatibility issue that is causing tf1 pytest to fail -- checking the tf1 and tf2 tests with the change locally now.

Reply:
With the change, the tests in tests/test_docker/test_dataset.py pass locally on the tf1, tf2, and pytorch containers

Reply:
Updated.